src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -925,11 +925,43 @@ private void ScanSyntaxToken(ref TokenInfo info)
                         info.Text = TextWindow.GetText(intern: true);
                     }
 
-                    this.AddError(ErrorCode.ERR_UnexpectedCharacter, info.Text);
+                    this.AddError(ErrorCode.ERR_UnexpectedCharacter, EscapeIfInvalidUnicode(info.Text));
```

Comment:
i played around for a bit, and this appeared to be the only place that needed patching up.  I'm hoping we can limit the spot fixing to only here.